code/modules/surgery/organs/external/tails.dm

```diff
@@ -34,13 +34,23 @@
 /obj/item/organ/external/tail/Remove(mob/living/carbon/organ_owner, special, moving)
 	if(wag_flags & WAG_WAGGING)
 		wag(FALSE)
+
+	return ..()
+
+/obj/item/organ/external/tail/on_remove(mob/living/carbon/organ_owner, special)
 	. = ..()
+
 	UnregisterSignal(organ_owner, COMSIG_ORGAN_WAG_TAIL)
 
 	if(type in organ_owner.dna.species.external_organs)
 		organ_owner.add_mood_event("tail_lost", /datum/mood_event/tail_lost)
 		organ_owner.add_mood_event("tail_balance_lost", /datum/mood_event/tail_balance_lost)
 
+	if(bodypart_overlay.color_source == NONE)
+		return
+
+	color = bodypart_overlay.draw_color // so a pink felinid doesn't drop a gray tail
```

Comment:
Can you move this up to the /external type? Snouts, horns, spines, podhair etc would also benefit from not being gray